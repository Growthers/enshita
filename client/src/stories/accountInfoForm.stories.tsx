// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
