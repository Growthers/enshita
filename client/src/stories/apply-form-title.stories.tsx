import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
