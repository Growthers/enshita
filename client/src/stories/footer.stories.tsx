// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
