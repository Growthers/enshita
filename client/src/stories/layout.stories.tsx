import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
