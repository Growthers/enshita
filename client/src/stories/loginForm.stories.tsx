// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
