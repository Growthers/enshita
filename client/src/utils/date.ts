export const fillzero = (num: number, digit: number) =>
  `${"0".repeat(digit)}${num}`.slice(-1 * digit);

// ISO8601形式の文字列を日付型に変換
export const str2Date = (str: string) => {
  // 19700101T090000+0900 の可能性があるので
  // 1970-01-01T09:00:00+09:00への変換

  try {
    const data = str.split("T");
    const isUTC = str.endsWith("Z");

    const strDate = data[0].split("-").join("");
    const strTime = data[1].split(":").join("").split("+").join("");

    const year = strDate.slice(0, 4);
    const month = strDate.slice(4, 6);
    const day = strDate.slice(6, 8);
    const hour = strTime.slice(0, 2);
    const minute = strTime.slice(2, 4);
    const second = strTime.slice(4, 6);

    const timezoneHour = isUTC ? "" : strTime.slice(6, 8);
    const timezoneMinute = isUTC ? "" : strTime.slice(8, 10);
    const timezone = isUTC ? "Z" : `+${timezoneHour}:${timezoneMinute}`;

    const datetime = `${year}-${month}-${day}T${hour}:${minute}:${second}${timezone}`;

    return new Date(datetime);
  } catch (e) {
    return null;
  }
};

export const getDateInfo = (date: Date) => {
  const year = date.getFullYear();
  const month = date.getMonth() + 1;
  const day = date.getDate();
  const hour = date.getHours();
  const minute = date.getMinutes();
  const second = date.getSeconds();
  const dayOfWeek = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"][
    date.getDay()
  ];

  const data = {
    year,
    month,
    day,
    hour,
    minute,
    second,
    dayOfWeek,
    strYear: fillzero(year, 4),
    strMonth: fillzero(month, 2),
    strDay: fillzero(day, 2),
    strHour: fillzero(hour, 2),
    strMinute: fillzero(minute, 2),
    strSecond: fillzero(second, 2),
  };

  return data;
};

export const getUNIXTime = (date: Date) => Math.floor(date.getTime() / 1000);

export const getDateDiff = (date1: Date, date2: Date) => {
  const diff = Math.abs(getUNIXTime(date1) - getUNIXTime(date2));

  // diffを日にち、時間、分、秒に変換する
  const second = diff % 60;
  const minute = Math.floor((diff % 3600) / 60);
  const hour = Math.floor((diff % 86400) / 3600);
  const day = Math.floor(diff / 86400);

  const data = {
    diff,
    second,
    minute,
    hour,
    day,
  };

  return data;
};
