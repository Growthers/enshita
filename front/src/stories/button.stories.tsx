import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
