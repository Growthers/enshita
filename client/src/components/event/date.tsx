import React from "react";

import { str2Date, getDateInfo, getDateDiff, fillzero } from "~/utils/date";
import styles from "~/styles/components/event/date.module.scss";

type Props = {
  startStrDate: string;
  endStrDate: string;
};

const EventDate: React.FC<Props> = props => {
  const { startStrDate, endStrDate } = props;
  const startDate = str2Date(startStrDate);
  const endDate = str2Date(endStrDate);

  if (startDate === null || endDate === null)
    return (
      <div className={styles["enshita-event-date"]}>
        <p>No Date</p>
      </div>
    );

  const startDateInfo = getDateInfo(startDate);
  const endDateInfo = getDateInfo(endDate);

  const nextDate = new Date(
    startDateInfo.year,
    startDateInfo.month - 1,
    startDateInfo.day,
  );
  nextDate.setDate(nextDate.getDate() + 1);
  const { diff } = getDateDiff(startDate, endDate);
  const { diff: nextDayDiff } = getDateDiff(startDate, nextDate);

  // 翌日の朝4時までにイベントが終わるとき
  if (diff - nextDayDiff <= 3600 * 4) {
    const endHour =
      startDateInfo.hour <= endDateInfo.hour
        ? endDateInfo.hour
        : 24 + endDateInfo.hour;
    return (
      <div className={styles["enshita-event-date"]}>
        <time>{`${startDateInfo.strYear}/${startDateInfo.strMonth}/${startDateInfo.strDay} (${startDateInfo.dayOfWeek})`}</time>
        <time>
          <span>{`${startDateInfo.strHour}:${
            startDateInfo.strMinute
          } ~ ${fillzero(endHour, 2)}:${endDateInfo.strMinute}`}</span>
        </time>
      </div>
    );
  }

  return (
    <div className={styles["enshita-event-date"]}>
      <time>
        <span>
          {`${startDateInfo.strYear}/${startDateInfo.strMonth}/${startDateInfo.strDay} (${startDateInfo.dayOfWeek}) ${startDateInfo.strHour}:${startDateInfo.strMinute} ~`}
        </span>
      </time>
      <time>
        <span>{`${endDateInfo.strYear}/${endDateInfo.strMonth}/${endDateInfo.strDay} (${endDateInfo.dayOfWeek}) ${endDateInfo.strHour}:${endDateInfo.strMinute}`}</span>
      </time>
    </div>
  );
};

export default EventDate;
