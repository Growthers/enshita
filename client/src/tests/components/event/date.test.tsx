import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/event/date.stories";

const { Default } = composeStories(stories);

describe("(component) EventDate", () => {
  test("Snap Shot - 19:30 ~ 23:00", () => {
    const { container } = render(
      <Default
        startStrDate="2022-08-28T19:30:00+09:00"
        endStrDate="2022-08-28T23:00:00+09:00"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - 19:30 ~ Next Day 04:00 ", () => {
    const { container } = render(
      <Default
        startStrDate="2022-08-28T19:30:00+09:00"
        endStrDate="2022-08-29T04:00:00+09:00"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - 19:30 ~ Next Day 05:00", () => {
    const { container } = render(
      <Default
        startStrDate="2022-08-28T19:30:00+09:00"
        endStrDate="2022-08-29T05:00:00+09:00"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - No Date", () => {
    const { container } = render(<Default startStrDate="" endStrDate="" />);
    expect(container).toMatchSnapshot();
  });
});
