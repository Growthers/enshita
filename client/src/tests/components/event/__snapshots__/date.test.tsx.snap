// Vitest Snapshot v1

exports[`(component) EventDate > Snap Shot - 19:30 ~ 23:00 1`] = `
<div>
  <div
    class="_enshita-event-date_06bc2d"
  >
    <time>
      2022/08/28 (Sun)
    </time>
    <time>
      <span>
        19:30 ~ 23:00
      </span>
    </time>
  </div>
</div>
`;

exports[`(component) EventDate > Snap Shot - 19:30 ~ Next Day 04:00  1`] = `
<div>
  <div
    class="_enshita-event-date_06bc2d"
  >
    <time>
      2022/08/28 (Sun)
    </time>
    <time>
      <span>
        19:30 ~ 28:00
      </span>
    </time>
  </div>
</div>
`;

exports[`(component) EventDate > Snap Shot - 19:30 ~ Next Day 05:00 1`] = `
<div>
  <div
    class="_enshita-event-date_06bc2d"
  >
    <time>
      <span>
        2022/08/28 (Sun) 19:30 ~
      </span>
    </time>
    <time>
      <span>
        2022/08/29 (Mon) 05:00
      </span>
    </time>
  </div>
</div>
`;

exports[`(component) EventDate > Snap Shot - No Date 1`] = `
<div>
  <div
    class="_enshita-event-date_06bc2d"
  >
    <p>
      No Date
    </p>
  </div>
</div>
`;
